packages/mobile/src/geth/GethNativeBridgeSigner.ts

```diff
@@ -0,0 +1,110 @@
+import {
+  encodeTransaction,
+  extractSignature,
+  RLPEncodedTx,
+  rlpEncodedTx,
+} from '@celo/contractkit/lib/utils/signing-utils'
+import { Signer } from '@celo/contractkit/lib/wallets/signers/signer'
+import { ensureLeading0x } from '@celo/utils/lib/address'
+import { normalizeAddressWith0x } from '@celo/utils/src/address'
+import * as ethUtil from 'ethereumjs-util'
+import { GethNativeModule } from 'react-native-geth'
+import Logger from 'src/utils/Logger'
+import { Tx } from 'web3-core'
+
+const INCORRECT_PASSWORD_ERROR = 'could not decrypt key with given password'
+const currentTimeInSeconds = () => Math.floor(Date.now() / 1000)
+
+const TAG = 'geth/GethNativeBridgeSigner'
+/**
+ * Implements the signer interface on top of a RNGeth instance
+ */
+export class GethNativeBridgeSigner implements Signer {
+  /**
+   * Construct a new instance of the RPC signer
+   *
+   * @param geth the RNGeth instance
+   * @param account Account address derived from the private key to be called in init
+   * @param unlockBufferSeconds Number of seconds to shrink the unlocked duration by to account for
+   * latency and timing inconsistencies on the node
+   * @param unlockTime Timestamp in seconds when the signer was last unlocked
+   * @param unlockDuration Number of seconds that the signer was last unlocked for
+   */
+  constructor(
+    protected geth: GethNativeModule,
+    protected account: string,
+    protected unlockBufferSeconds = 5,
+    protected unlockTime = -1,
+    protected unlockDuration = -1
```

Comment:
These are all in seconds, but are they required to be integral? or can you pass numbers with a decimal part?

Reply:
This functionality is pretty much copied from what's happening in the `RpcSigner`, and by looking at the behaviour I feel like using sub-second values here would result in unexpected behaviour because when verifying duration the current time is `floored` to the second.